File: js/euler77.js
function euler76(){
    const sieved = new Array(101).fill(false);
    const sums = new Array(101).fill(0);
    
    for (let i = 2; i < 100; i++){
        if (!sieved[i]){
            sums[i]++;
            for (let j = 2 * i; j < sieved.length; j += i){
                sieved[j] = true;
            }
            for (let n = i + 1; n < sums.length; n++){
                sums[n] += sums[n - i];
            }
        }
    }
    for (let i = 0; i < sums.length; i++){
        if (sums[i] > 5000) return i;
    }
}

console.log(euler76());

File: go/euler77.go
package main
import "fmt"

func sieve(n int) []int {
    primes := []int{}
    sieved := make([]bool, n + 1)
    for i := 2; i <= n; i++ {
        if !sieved[i] {
            primes = append(primes, i)
            for j := 2 * i; j <= n; j += i {
                sieved[j] = true
            }
        }
    }
    return primes
}

func main() {
    primes := sieve(100)
    sums := [101]int{}
    sums[0] = 1
    for _, i := range primes {
        for j := i; j < len(sums); j++ {
            sums[j] += sums[j - i]
        }
    }
    for i := 0; i < len(sums); i++ {
        if sums[i] > 5000 {
            fmt.Println(i)
            break
        }
    }
}
